### DISCUSSION TITLE
Facing Exception upon Installation of Vendure

### DISCUSSION THREAD

**Discussion by ahmerdevopsinertia** [vendurehq/vendure#607] [Q&A]
I am facing this error while installing the Vendure on my Linux machine. Please guide me to resolve the issue

**error 1/1/21, 2:59 AM - [ExceptionHandler] Cannot use GraphQLSchema "[object GraphQLSchema]" from another module or realm**

**Comment by michaelbromley** [comment]
Hi,

Can you detail the exact steps you took to install? E.g. did you use the `@vendure/create` installer or manually add via npm/yarn?

**Reply by ahmerdevopsinertia** [reply]
Thanks for the response @michaelbromley, I have tried with following commands.
npx @vendure/create my-app
npm init @vendure my-app

**Reply by michaelbromley** [reply]
OK. Can you post the full output of the `npx @vendure/create my-app` command?

**Reply by ahmerdevopsinertia** [reply]
ok just wait

**Reply by ahmerdevopsinertia** [reply]
![vendure_create_error](https://user-images.githubusercontent.com/58416271/105459710-875bc200-5cac-11eb-895a-c54fc1efd4ea.PNG)

@michaelbromley  Find the attached Error log

**Reply by michaelbromley** [reply]
Thanks. So it says there are multiple versions of GraphQL installed. You can check this by running `yarn why graphql` from the project root. What's the output of that?

**Reply by ahmerdevopsinertia** [reply]
> Thanks. So it says there are multiple versions of GraphQL installed. You can check this by running `yarn why graphql` from the project root. What's the output of that?

Thanks for the response @michaelbromley. Here is the output of yarn why graphql
![graphql_output](https://user-images.githubusercontent.com/58416271/105570950-24326400-5d6e-11eb-9a14-7165f3226bb0.PNG)

**Reply by ahmerdevopsinertia** [reply]
@michaelbromley Additionally, can you confirm where Vendure puts all the error logs regarding the email plugin. As am not getting the email upon completing order the mail box server outputs something on this location "**/static/email/test-emails/**" but not send and email. So i have to check the logs where the email is stucking. Here is my config for the email

```
EmailPlugin.init({
            devMode: true,
            outputPath: path.join(__dirname, '../static/email/test-emails'),
            mailboxPort: 3003,
            handlers: defaultEmailHandlers,
            templatePath: path.join(__dirname, '../static/email/templates'),
            globalTemplateVars: {
                // The following variables will change depending on your storefront implementation
                fromAddress: '"example" <noreply@example.com>',
                verifyEmailAddressUrl: 'http://localhost:8080/verify',
                passwordResetUrl: 'http://localhost:8080/password-reset',
                changeEmailAddressUrl: 'http://localhost:8080/verify-email-address-change'
            },
            transport: {
                type: 'smtp',
		secure: false,
                host: 'demo-custom-ecommerce.XXXXX.XXX',
                port: 25,
                auth: {
                  user: 'admin@demo-custom-ecommerce.XXXXX.XXX',
                  pass: 'XXXXXXXXXXX',
                },
		tls: {
      		   rejectUnauthorized: false
                },
              	logging: true,
	        debug: true,
            },
        }),
```

**Reply by michaelbromley** [reply]
The output of `yarn why graphql` looks fine, just like the installation I just tested, which worked without issue. I don't have enough information to advise further on that point, I'm afraid.

Regarding the emails, the logs are output to the configured VendureLogger (which is console by default). However, if you emails are going to the `test-emails` folder it means the email plugin is running with `devMode: true`, which suggests that the config above is for some reason not being used by the server. Could it be that you need to transpile to js or something?

**Reply by ahmerdevopsinertia** [reply]
@michaelbromley Thanks for the response it is really appreciated. Actually am using this Vendure Back-End in one of my front-end React App. This React App depends upon the apollo-graphql package. So I think GraphQL of Vendure and this React is conflicting as my React App and Vendure Back-End is hosted on the same machine for now.

**Reply by ahmerdevopsinertia** [reply]
@michaelbromley Thanks Emails are working fine after removing the **devMode: true**.

**Reply by ahmerdevopsinertia** [reply]
@michaelbromley Can you help me in setting up the **Braintree** sandbox payment in my checkout process.

**Reply by ahmerdevopsinertia** [reply]
@michaelbromley Can you help me in setting up the Braintree sandbox payment in my checkout process.

**Reply by michaelbromley** [reply]
I suggest you refer to the Braintree developer docs for that. I like to help but I must limit my support to Vendure-specific issues at the moment.

**Reply by ahmerdevopsinertia** [reply]
Got it

**Reply by ahmerdevopsinertia** [reply]
Thanks alot

**Comment by shijupoothotta** [comment]
![image](https://user-images.githubusercontent.com/2807290/119058387-ed9eea00-b9eb-11eb-8538-a8794ee53429.png)

**Reply by shijupoothotta** [reply]
Getting this error in Linux - Centos.
yarn why graphql 
Output is given below.
![image](https://user-images.githubusercontent.com/2807290/119058537-348cdf80-b9ec-11eb-9a33-99c9658f2ddd.png)

**Reply by shijupoothotta** [reply]
npm install output. If if throw some info
![image](https://user-images.githubusercontent.com/2807290/119058894-dd3b3f00-b9ec-11eb-96dd-e463858197a8.png)

**Comment by dong-nz** [comment]
Hi team,

I got the same issue while creating app.

Stacktrace:

vendure npx @vendure/create my-app

Welcome to @vendure/create v1.0.0!

Let's configure a new Vendure project. First a few questions:

✔ Which database are you using? › SQLite
✔ Which programming language will you be using? › TypeScript
✔ Populate with some sample product data? … no / yes
✔ What identifier do you want to use for the superadmin user? … superadmin
✔ What password do you want to use for the superadmin user? … superadmin

Setting up your new Vendure project in /Users/dong.nguyen/Documents/dong-dev/vendure/my-app
This may take a few minutes...

  ✔ Installing dependencies
  ✔ Generating app scaffold
  ⠸ Initializing server
error 5/25/21, 8:37 AM - [ExceptionHandler] Cannot use GraphQLSchema "[object GraphQLSchema]" from another module or realm.

Ensure that there is only one instance of "graphql" in the node_modules
directory. If different versions of "graphql" are the dependencies of other
relied on modules, use "resolutions" to ensure only one version is installed.

https://yarnpkg.com/en/docs/selective-version-resolutions

Duplicate "graphql" modules cannot be used at the same time since different
versions may have different capabilities and behavior. The data from one
version used in the function from another could produce confusing and
spurious results.
Error: Cannot use GraphQLSchema "[object GraphQLSchema]" from another module or realm.

Ensure that there is only one instance of "graphql" in the node_modules
directory. If different versions of "graphql" are the dependencies of other
relied on modules, use "resolutions" to ensure only one version is installed.

https://yarnpkg.com/en/docs/selective-version-resolutions

Duplicate "graphql" modules cannot be used at the same time since different
versions may have different capabilities and behavior. The data from one
version used in the function from another could produce confusing and
spurious results.
    at instanceOf (/Users/dong.nguyen/Documents/dong-dev/vendure/my-app/node_modules/graphql/jsutils/instanceOf.js:28:13)
    at Object.isSchema (/Users/dong.nguyen/Documents/dong-dev/vendure/my-app/node_modules/graphql/type/schema.js:36:34)
    at /Users/dong.nguyen/Documents/dong-dev/vendure/my-app/dist/stitch/src/stitchSchemas.js:29:14
    at Array.forEach (<anonymous>)
    at Object.stitchSchemas (/Users/dong.nguyen/Documents/dong-dev/vendure/my-app/dist/stitch/src/stitchSchemas.js:28:13)
    at Object.generatePermissionEnum (/Users/dong.nguyen/Documents/dong-dev/vendure/my-app/node_modules/@vendure/core/src/api/config/generate-permissions.ts:38:12)
    at buildSchemaForApi (/Users/dong.nguyen/Documents/dong-dev/vendure/my-app/node_modules/@vendure/core/src/api/config/configure-graphql-module.ts:144:18)
    at createGraphQLOptions (/Users/dong.nguyen/Documents/dong-dev/vendure/my-app/node_modules/@vendure/core/src/api/config/configure-graphql-module.ts:92:25)

**Reply by michaelbromley** [reply]
Can you tell me which version of npm you are using?

**Reply by dong-nz** [reply]
Hi Michael,

I used to latest node & npm version. Thank you for your support.

{
  npm: '7.14.0',
  node: '14.16.1'
}

**Reply by michaelbromley** [reply]
I think this might be an issue with changes introduces in npm v7. If you can, try with npm v6.x and let me know if that works.

**Reply by dong-nz** [reply]
yes, it works for npm v6.x.

**Reply by michaelbromley** [reply]
OK, thanks for confirming. I need to make some changes then in order for it to work with v7.x

**Reply by michaelbromley** [reply]
This should now be fixed with the release of v1.0.1 👍
